zipkin-lens/src/components/DependenciesPage/DependenciesGraph.tsx

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+/* eslint-disable no-shadow */
+import React, { CSSProperties } from 'react';
+import ReactSelect, { ValueType } from 'react-select';
+import { AutoSizer } from 'react-virtualized';
+import { Box, Grid, useTheme } from '@material-ui/core';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import moment from 'moment';
+
+import Dependencies from '../../types/Dependencies';
+import VizceralWrapper from './VizceralWrapper';
+import NodeDetailData from './NodeDetailData';
+import { Edge } from './types';
+
+// These filter functions use any type because they are passed directly to untyped JS code.
+const filterConnections = (object: any, value: any) => {
+  if (!value) {
+    return true;
+  }
+  if (object.name === value) {
+    return true;
+  }
+  return object.source.name === value || object.target.name === value;
+};
+
+const filterNodes = (object: any, value: any) => {
+  if (!value) {
+    return true;
+  }
+  if (object.name === value) {
+    return true;
+  }
+  return (
+    object.incomingConnections.find(
+      (conn: any) => conn.source.name === value,
+    ) ||
+    object.outgoingConnections.find((conn: any) => conn.target.name === value)
+  );
+};
+
+const reactSelectStyles = {
+  control: (base: CSSProperties) => ({
+    ...base,
+    width: '15rem',
+  }),
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    containerGrid: {
+      height: '100%',
+    },
+    itemGrid: {
+      height: '100%',
+    },
+    searchButton: {
+      fontSize: '1.2rem',
+      padding: theme.spacing(1),
+      minWidth: 0,
+      width: 32,
+      height: 32,
+    },
+  }),
+);
+
+interface Props {
+  dependencies: Dependencies;
+}
+
+const DependenciesGraph: React.FC<Props> = ({ dependencies }): JSX.Element => {
```

Comment:
Don't think we need the return type `JSX.Element` for components since `React.FC` defines a return type.

Reply:
OK!